### PR TITLE
Zero Networks v1.0.0

### PR DESCRIPTION
**Pull request by dicolanl**
Required items, please complete
   
   Change(s):
   - Created Azure Function based data connector
   - Added Zero Networks Logo
   - Created Custom Connector and 3 Playbooks
   - Created sample data
   - Created Solution:
3 Analytics Rules
Native Poller Data Collector
4 Hunting Queries
Parser
Playbook (linked deployments)
Workbook.

   Reason for Change(s):
   - New integration with Zero Networks

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by vmanojreddy** [comment]
@aprakash13 Please review analytic rules. Thanks.

File: Solutions/ZeroNetworks/Parsers/ZNAccessOrchestratorAudit.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    3, "Access Portal",
+    4, "Admin Portal",
+    5, "AI",
+    6, "API"
+];
+let UserRoleTypeTable = datatable (userRole_d: double, UserRole: string) [
+    1, "Admin",
+    2, "Viewer",
+    3, "Regular",
+    4, "API - Full Access"
```
Comment (vmanojreddy) [inline Solutions/ZeroNetworks/Parsers/ZNAccessOrchestratorAudit.txt:54]: @dicolanl Please add the comma at the end of line

**Review by vmanojreddy — CHANGES_REQUESTED** [review]

**Comment by vmanojreddy** [comment]
@anki-narravula Please review the Playbooks. thanks!!!

File: Solutions/ZeroNetworks/Parsers/ZNAccessOrchestratorAudit.txt (hunk 1 above)
Comment (dicolanl) [reply Solutions/ZeroNetworks/Parsers/ZNAccessOrchestratorAudit.txt]: Fixed.

**Comment by vmanojreddy** [comment]
@dicolanl In sample data, the columns timestamp_d,auditType_d,enforcementSource_d,userRole_d have the numeric value but as you have added " " to values they are been saved with _s rather than _d , please check and update, thanks.

**Comment by vmanojreddy** [comment]
@dicolanl In checks list Data connector validations failing due to DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json does not match with Azure Function Connector Template.
Path reference : https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json

**Comment by vmanojreddy** [comment]
@aprakash13 Please review analytic rules. Thanks!!!

File: DataConnectors/ZeroNetworks/AccessOrchestratorFunctionConnector/DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+						],
+						"label": "Primary Key"
+					},
+					"type": "CopyableLabel"
+				}
+			]
+		},
+		{
+			"title": "Option 1 - Azure Resource Manager (ARM) Template",
+			"description": "Use this method for automated deployment of the Zero Networks Access Orchestrator Audit data connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FAzure-Sentinel%2Fmaster%2FDataConnectors%2FZeroNetworks%2FAccessOrchestratorFunctionConnector%2Fazuredeploy_Connector_ZeroNetworks_AccessOrchestrator_Audit_AzureFunction.json)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n> **NOTE:** Within the same resource group, you can't mix Windows and Linux apps in the same region. Select existing resource group without Windows apps in it or create new resource group.\n3. Enter the **APIToken** and deploy. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
```
Comment (vmanojreddy) [inline DataConnectors/ZeroNetworks/AccessOrchestratorFunctionConnector/DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json:109]: @dicolanl  Please update the azuredeploy short link https://aka.ms/sentinel-ZeroNetworks-azuredeploy

**Review by vmanojreddy — CHANGES_REQUESTED** [review]

File: DataConnectors/ZeroNetworks/AccessOrchestratorFunctionConnector/DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json (hunk 2 above)
Comment (dicolanl) [reply DataConnectors/ZeroNetworks/AccessOrchestratorFunctionConnector/DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json]: Done

**Comment by dicolanl** [comment]
> vmanojreddy

This error

> @dicolanl In checks list Data connector validations failing due to DataConnector_API_AzureFunctionApp_ZeroNetworks_AccessOrchestrator_Audit.json does not match with Azure Function Connector Template. Path reference : https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json

Fixed!

**Comment by vmanojreddy** [comment]
@anki-narravula Please review the Playbooks. thanks!!!

**Comment by vmanojreddy** [comment]
@aprakash13 Please review analytic rules. Thanks!!!

**Comment by vmanojreddy** [comment]
> > @dicolanl Please move the files into solution folder. thanks!!!
> 
> @vmanojreddy what files exactly? most are already in solution folder, except for a few things like logo, function app connector and playbooks.

@dicolanl  Yes, Can You please move  function app connector and playbooks into solution folder. thanks

**Comment by dicolanl** [comment]
> > > @dicolanl Please move the files into solution folder. thanks!!!
> > 
> > 
> > @vmanojreddy what files exactly? most are already in solution folder, except for a few things like logo, function app connector and playbooks.
> 
> @dicolanl Yes, Can You please move function app connector and playbooks into solution folder. thanks

@vmanojreddy done

**Comment by vmanojreddy** [comment]
@aprakash13 Please review analytic rules. Thanks!!!

**Comment by vmanojreddy** [comment]
@dicolanl In Data connector zip folder should not be in correct format. Can you please check once. thanks!!!

**Comment by vmanojreddy** [comment]
@dicolanl In Data connector zip folder should not be in correct format. Can you please check once. thanks!!!

**Review by aprakash13 — APPROVED** [review]
Looks good from a query perspective.

**Comment by NikTripathi** [comment]
Hey @dicolanl ,
Sorry for responding late.
I saw that the zip folder we are using here is not in a correct format. We need to change the structure for that. Inside zip folder there is another folder and then contents. It should be contents directly inside the zip folder. We don't use another folder inside zip. That is causing some deployment issues.

Thanks
Nikhil

**Comment by NikTripathi** [comment]
Hi @dicolanl Also after deploying the data connector I am not able to get the function. I checked the .zip folder and I found out that there are no function added in zip folder. Can you please update the zip package with all the required content. Here is a zip package link for you reference. https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Cloudflare/Data%20Connectors/CloudflareConn.zip

Thanks
Nikhil

**Comment by vmanojreddy** [comment]
@rushriva Please review the Playbooks and provide your sign. thanks!!!

**Review by sreedharande — DISMISSED** [review]
Reviewed Playbooks

**Review by NikTripathi — CHANGES_REQUESTED** [review]
Hi @dicolanl,
Functions are not loading. Can you please look into it?
image

**Comment by dicolanl** [comment]
> Hi @dicolanl,
> Functions are not loading. Can you please look into it?
> image

This will not work until it’s merged becuase the url for the zip files doesn’t actually exist until it’s merged in.

**Comment by NikTripathi** [comment]
> > Hi @dicolanl,
> > Functions are not loading. Can you please look into it?
> > image
> 
> This will not work until it’s merged becuase the url for the zip files doesn’t actually exist until it’s merged in.

Hi @dicolanl 
After updating the link it is still not working. I used the PR zip link in configuration.

**Comment by dicolanl** [comment]
> > > Hi @dicolanl,
> > > Functions are not loading. Can you please look into it?
> > > image
> > 
> > 
> > This will not work until it’s merged becuase the url for the zip files doesn’t actually exist until it’s merged in.
> 
> Hi @dicolanl After updating the link it is still not working. I used the PR zip link in configuration.

@NikTripathi  try now.  found an issue with the zip.  also fixed an issue with storage account naming.
image

**Comment by dicolanl** [comment]
@NikTripathi @vmanojreddy @preetikr 
What else is needed to merge this??

**Review by sreedharande — APPROVED** [review]
All requested changes has been Addressed

**Comment by dicolanl** [comment]
> Hi @dicolanl can you please confirm us why we are using two types connectors function app based connector and ccp connector?

@vmanojreddy because according to [this](https://docs.microsoft.com/en-us/azure/sentinel/create-codeless-connector?tabs=deploy-via-arm-template%2Cconnect-via-the-azure-portal) CCP is in preview.  Some customers will be unwilling to use a preview feature so I created the azure function model as this is supported.

**Pull request merged**